terminus-ui/paginator/src/paginator.component.spec.ts

```diff
@@ -1,470 +1,309 @@
-        value: 2,
-      };
-      component.pageSelect.emit = jest.fn();
-      component['createCurrentPageLabel'] = jest.fn();
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-
-      expect(component.currentPageIndex).toEqual(0);
-
-      component.currentPageChanged(eventMock);
-
-      expect(component.currentPageIndex).toEqual(2);
-      expect(component['createCurrentPageLabel']).toHaveBeenCalled();
-      expect(component.pageSelect.emit).toHaveBeenCalled();
-    });
-
-  });
-
-
-  describe(`changePage()`, () => {
-
-    test(`should fake a changed page event when valid`, () => {
-      jest.spyOn(component, 'currentPageChanged');
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      component.changePage(2, 1, component.pagesArray);
-
-      expect(component.currentPageChanged).toHaveBeenCalled();
-    });
-
-
-    test(`should do nothing when the destination is not valid`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      jest.spyOn(component, 'currentPageChanged');
-
-      component.changePage(-1, 1, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
-
-      const invalidLength = component.pagesArray.length + 2;
-      component.changePage(invalidLength, 1, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
-    });
-
-
-    test(`should do nothing when already on the requested page`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      jest.spyOn(component, 'currentPageChanged');
+    test(`should set and retrieve`, () => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      fixture.detectChanges();
 
-      component.changePage(2, 2, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
+      expect(hostComponent.showRecordsPerPageSelector).toEqual(true);
+      expect(fixture.debugElement.query(By.css('.qa-paginator-per-page-select'))).toBeTruthy();
     });
 
-  });
-
-
-  describe(`isFirstPage()`, () => {
+    test(`should change page on select`, () => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      const index: string = hostComponent.recordsPerPageChoices[1].toString();
+      fixture.detectChanges();
+      updateRecordsPerPage(fixture, index);
 
-    test(`should return TRUE when the passed in page is 0`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+      expect(titleEl.textContent).toContain('1 - 20 of 100');
 
-      expect(component.isFirstPage(0)).toEqual(true);
     });
 
+    test(`should be hidden when set`, () => {
```

Comment:
I chose to separate them because there are two things required for the drop down to show. Maybe once we remove the logic from the template, we can combine them.
```html
    <ts-select
      class="qa-paginator-per-page-select"
      *ngIf="recordsPerPageChoices && showRecordsPerPageSelector"
```